entity-server-api/src/main/java/org/terracotta/entity/ServiceProvider.java

```diff
@@ -22,46 +22,69 @@
 
 
 /**
- * Service Provider which are provided at the platform level.  A service provider is responsible for instantiating the
- * services to interact with its underlying implementation in response to platform requests.  While the services are
- * generally used by entities, they are also used by the platform, itself.
+ * A service provider is responsible for instantiating the services to interact with its underlying implementation in response
+ *  to platform requests.  While the services are generally used by entities, they are also used by the platform, itself.
+ * 
+ * Note that there are no threading restrictions on how the methods in this interface will be called.  This means that any
+ *  implementation which has shared state must ensure that it can manage this state in a thread safe way (subject to whatever
+ *  thread safety its own state requires).
  */
 public interface ServiceProvider {
+  /**
+   * PLATFORM_CONSUMER_ID is the consumerID used by the platform, itself, when it wants to interact with services.  Normal
+   *  entity instances use positive consumeID values.
+   */
+  public final long PLATFORM_CONSUMER_ID = 0; 
 
   /**
-   * The platform configuration based on which the Service provider can choose to initialize itself.
-   *
-   * @param configuration platform configuration
-   * @return true if provider has successfully been initialized with the provided configuration
+   * The platform configuration based on which the receiver can choose to initialize itself.
+   * This is called once on each provider instance, at server start-up time, before any other methods have been called on the
+   *  instance.
+   * Note that it is possible for there to be multiple instances of the same ServiceProvider if the server had multiple
+   *  configuration stanzas handled by the same ServiceProvider class.  As the entities requesting services do not have direct
+   *  access to the set of initialized ServiceProviders, they do not require special accounting for this possibility.
```

Comment:
I also find this comment on the `initialize` method to be a bit weird. Since that is lifecycle handled outside of this method and others here, this should move to class level javadoc. And there you could also explain how different `ServiceProvider` will be invoked one after another in case a specific one cannot provide its service because the `ServiceConfiguration` does not match.